src/serac/numerics/equation_solver.cpp

```diff
@@ -88,133 +212,41 @@ std::unique_ptr<mfem::AmgXSolver> configureAMGX(const MPI_Comm comm, const AMGXP
 
   return amgx;
 }
-
 #endif
-}  // namespace detail
 
-std::unique_ptr<mfem::IterativeSolver> EquationSolver::BuildIterativeLinearSolver(
-    MPI_Comm comm, const IterativeSolverOptions& lin_options)
+std::unique_ptr<mfem::Solver> buildPreconditioner(Preconditioner preconditioner, int print_level,
+                                                  [[maybe_unused]] MPI_Comm comm)
 {
-  std::unique_ptr<mfem::IterativeSolver> iter_lin_solver;
-
-  switch (lin_options.lin_solver) {
-    case LinearSolver::CG:
-      iter_lin_solver = std::make_unique<mfem::CGSolver>(comm);
-      break;
-    case LinearSolver::GMRES:
-      iter_lin_solver = std::make_unique<mfem::GMRESSolver>(comm);
-      break;
-    case LinearSolver::MINRES:
-      iter_lin_solver = std::make_unique<mfem::MINRESSolver>(comm);
-      break;
-    default:
-      SLIC_ERROR_ROOT("Linear solver type not recognized.");
-      exitGracefully(true);
-  }
-
-  iter_lin_solver->SetRelTol(lin_options.rel_tol);
-  iter_lin_solver->SetAbsTol(lin_options.abs_tol);
-  iter_lin_solver->SetMaxIter(lin_options.max_iter);
-  iter_lin_solver->SetPrintLevel(lin_options.print_level);
+  std::unique_ptr<mfem::Solver> preconditioner_ptr;
```

Comment:
There are several variables here that end in`_ptr`. We don't do that anywhere else in the code. We should be consistent unless there is a reason.